docker/Dockerfile.common

```diff
@@ -54,7 +56,32 @@ RUN apt-get -y install \
 RUN python3 -m pip install --upgrade pip
 RUN python3 -m pip install aiofiles debugpy typing_extensions typing
 
+# avahi advertiser support
+RUN apt-get -y install \
+    libnss-mdns \
+    avahi-discover \
+    libavahi-compat-libdnssd-dev
+
 ENV SCRYPTED_DOCKER_SERVE="true"
 ENV SCRYPTED_CAN_RESTART="true"
 ENV SCRYPTED_VOLUME="/server/volume"
 ENV SCRYPTED_INSTALL_PATH="/server"
+
+# copy configurations and scripts
+COPY fs /
+
+# s6 process supervisor
+RUN case "$(uname -m)" in \
+    x86_64) S6_ARCH='x86_64';; \
+    armv7l) S6_ARCH='armhf';; \
+    aarch64) S6_ARCH='aarch64';; \
+    *) echo "Your system architecture isn't supported."; exit 1 ;; \
+    esac \
+  && cd /tmp \
+  && set -x \
+  && curl -SLOf https://github.com/just-containers/s6-overlay/releases/download/v${S6_OVERLAY_VERSION}/s6-overlay-noarch.tar.xz \
+  && tar -C / -Jxpf /tmp/s6-overlay-noarch.tar.xz \
+  && curl -SLOf  https://github.com/just-containers/s6-overlay/releases/download/v${S6_OVERLAY_VERSION}/s6-overlay-${S6_ARCH}.tar.xz \
+  && tar -C / -Jxpf /tmp/s6-overlay-${S6_ARCH}.tar.xz
+
+ENTRYPOINT ["/init"]
```

Comment:
This is the recommended install method for s6-overlay as detailed [here](https://github.com/just-containers/s6-overlay#customizing-s6-behaviour). The specific implementation is similar to [oznu's implementation](https://github.com/oznu/docker-homebridge/blob/0aa674ba65f15c3512df3a5e4cdef80d56281e1b/Dockerfile#L58-L69) due to s6-overlay only being available on specific architectures. Since s6-overlay is the process supervisor for this container, it is the init process (PID 1) and should be started with `ENTRYPOINT`.